NetworkConfigurations/NetworkDiscoveryAndManagementSubCategories/STP_SpanningTreeProtocol.md

```diff
@@ -0,0 +1,277 @@
+---
+title: STP (Spanning Tree Protocol)
+grand_parent: Network Configuration
+parent: Network Discovery and Management
+nav_order: 1
+layout: default
+---
+
+# STP (Spanning Tree Protocol)
+
+## Introduction
+
+In this guide, we will quickly explain what the
+Spanning Tree Protocol is
+and give an example implementing it.
+
+The Spanning Tree Protocol or STP is a network protocol that operates
+on the Layer 2 level of devices.
+
+The ports on a switch with STP enabled
+will always be in one of the following five port states.
+
+1. Blocking
+
+2. Listening
+
+3. Learning
+
+4. Forwarding
+
+5. Disabled
+
+The ports in a switch with STP will often begin
+in the blocking state
+and later change to the listening, learning,
+and eventually forwarding state when appropriate.
+
+**This conversion may take time as devices "learn"
+where other devices are.**
+
+STP is simple to implement using bridges and is
+important to address looping and
+wasting bandwidth. STP works by blocking
+redundant links from forwarding packets.
+
+To learn the basics of bridges, visit the following:
+[Bridging Layer 2](../VLANConfigurationSubCategories/BridgingL2.html)
+
+## Enabling Spanning Tree Protocol
+
+**By default, STP is disabled on all bridges.**
+
+To enable STP on an existing bridge, run the following command:
+
+```
+$ ip link set dev ${Bridge Name} type bridge stp_state 1
+```
+
+To enable STP on bridge creation, use:
+
+```
+$ ip link set dev ${Bridge Name} type bridge stp_state 1
+```
+
+To disable STP on a bridge, set stp_state to “0”.
+
+To show the current information about the state of the
+bridge use the following:
+
+```
+$ ip -d link show dev ${Bridge Name}
+```
+
+The brctl utility can also be used to view information
+about the current configuration.
+
+If the brctl utility is not already installed, use:
```

Comment:
please review what "apt-get update" and "apt-get upgrade" do, and update this paragraph.